Index: java/HBM/src/main/resources/log4j.xml
===================================================================
--- java/HBM/src/main/resources/log4j.xml	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/resources/log4j.xml	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
@@ -1,21 +0,0 @@
-<?xml version="1.0" encoding="UTF-8" ?>
-<!DOCTYPE log4j:configuration SYSTEM "log4j.dtd">
-<log4j:configuration debug="false">
-
-    <appender name="console" class="org.apache.log4j.ConsoleAppender">
-        <layout class="org.apache.log4j.PatternLayout">
-            <param name="ConversionPattern" value="%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n" />
-        </layout>
-    </appender>
-
-    <logger name="org.hibernate">
-        <level value="OFF"/>
-        <appender-ref ref="console" />
-    </logger>
-
-    <root>
-        <level value="DEBUG" />
-        <appender-ref ref="console" />
-    </root>
-
-</log4j:configuration>
\ No newline at end of file
Index: java/PTL/src/main/java/com/ptl/client/AitClient.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/PTL/src/main/java/com/ptl/client/AitClient.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/PTL/src/main/java/com/ptl/client/AitClient.java	(date 1562916536861)
@@ -47,7 +47,7 @@
             this.aitReceiverSender = new AitReceiverSender(s);
         } catch (IOException e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
@@ -70,7 +70,7 @@
                             aitReceiverSender.close();
                         } catch (IOException ex) {
                             try {
-                                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                                AitLogger.getInstance().logErrorToFile(e);
                             } catch (Exception exc) {
                                 ex.printStackTrace();
                                 exc.printStackTrace();
@@ -90,14 +90,14 @@
                     try {
                         aitReceiverSender.close();
                         try {
-                            AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                            AitLogger.getInstance().logErrorToFile(e);
                         } catch (Exception exc) {
                             e.printStackTrace();
                             exc.printStackTrace();
                         }
                     } catch (IOException ex) {
                         try {
-                            AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                            AitLogger.getInstance().logErrorToFile(e);
                         } catch (Exception exc) {
                             ex.printStackTrace();
                             exc.printStackTrace();
@@ -216,7 +216,7 @@
                         "Login send... " });
             } catch (Exception e) {
                 try {
-                    AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                    AitLogger.getInstance().logErrorToFile(e);
                 }catch (Exception ex) {
                     e.printStackTrace();
                     ex.printStackTrace();
Index: java/PTL/src/main/java/com/ptl/database/AitLocalDB.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/PTL/src/main/java/com/ptl/database/AitLocalDB.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/PTL/src/main/java/com/ptl/database/AitLocalDB.java	(date 1562916536839)
@@ -40,7 +40,7 @@
             return m_Instance;
         } catch (Exception e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
Index: java/PTL/src/main/java/com/ptl/server/AitClientHandler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/PTL/src/main/java/com/ptl/server/AitClientHandler.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/PTL/src/main/java/com/ptl/server/AitClientHandler.java	(date 1562916536808)
@@ -41,7 +41,7 @@
                     "com/ptl/client " + clientData.getId() + " connecting..." });
         } catch (Exception e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
@@ -55,7 +55,7 @@
                 msgProccessing(received);
             }catch (Exception e) {
                 try {
-                    AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                    AitLogger.getInstance().logErrorToFile(e);
                 } catch (Exception exc) {
                     e.printStackTrace();
                     exc.printStackTrace();
@@ -83,7 +83,7 @@
         }
         catch (Exception e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
@@ -244,7 +244,7 @@
         }
         catch (Exception e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
Index: java/PTL/src/main/java/com/ptl/managers/AitLogger.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/PTL/src/main/java/com/ptl/managers/AitLogger.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/PTL/src/main/java/com/ptl/managers/AitLogger.java	(date 1562918952609)
@@ -50,18 +50,39 @@
         }
     }
 
-    public void logToFile(Object[] params, String title) throws Exception {
-        logToFile(params, title, AitLoggerPriority.ERROR);
+    public void logInfoToFile(String message) {
+        try {
+            logToFile(null, message, null, AitLoggerPriority.INFORMATION);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void logErrorToFile(Exception exception) {
+        logErrorToFile(null, exception);
     }
 
-    private void logToFile(Object[] params, String title, AitLoggerPriority priority) throws Exception  {
-        createLogDir();
+    public void logErrorToFile(String msg, Exception exception) {
+        try {
+            logToFile(exception.getStackTrace(), exception.toString(), msg, AitLoggerPriority.ERROR);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
 
-        StringBuilder builder = new StringBuilder();
+    private void logToFile(Object[] params, String title, String message, AitLoggerPriority priority) throws Exception  {
+        String dicFolderName;
+        if(AitLoggerPriority.ERROR.equals(priority)) {
+            dicFolderName = "errors";
+        } else {
+            dicFolderName = "logs";
+        }
+        createDir(dicFolderName);
+        File file = createFile(dicFolderName);
 
-        File logFile = createLogFile();
-        FileReader fr = new FileReader(logFile.getAbsoluteFile());
+        FileReader fr = new FileReader(file.getAbsoluteFile());
         BufferedReader br = new BufferedReader(fr);
+        StringBuilder builder = new StringBuilder();
         String line;
 
         do {
@@ -72,18 +93,24 @@
             }
         } while(line != null);
 
+        if(null != message && !message.isEmpty()) {
+            builder.append(String.format(convertDateTimeToString() + convertPiorityToString(priority), message));
+            builder.append(System.getProperty("line.separator"));
+        }
         builder.append(String.format(convertDateTimeToString() + convertPiorityToString(priority), title));
         builder.append(System.getProperty("line.separator"));
 
-        for (Object arg : params) {
-            String msg = arg.toString();
-            builder.append(msg);
-            builder.append(System.getProperty("line.separator"));
-        }
-        builder.append(System.getProperty("line.separator"));
-        builder.append(System.getProperty("line.separator"));
+        if(params != null && params.length > 0) {
+            for (Object arg : params) {
+                String msg = arg.toString();
+                builder.append(msg);
+                builder.append(System.getProperty("line.separator"));
+            }
+            builder.append(System.getProperty("line.separator"));
+            builder.append(System.getProperty("line.separator"));
+        }
 
-        FileWriter fw = new FileWriter(logFile.getAbsoluteFile());
+        FileWriter fw = new FileWriter(file.getAbsoluteFile());
         BufferedWriter bw = new BufferedWriter(fw);
         bw.write(builder.toString());
         bw.close();
@@ -119,8 +146,8 @@
         }
     }
 
-    private void createLogDir() {
-        String logDirPath = m_MainPath + "\\Log";
+    private void createDir(String dicName) {
+        String logDirPath = m_MainPath + "\\" + dicName;
         File logDir = new File(logDirPath);
 
         if(!logDir.exists()) {
@@ -133,10 +160,10 @@
         }
     }
 
-    private File createLogFile() throws Exception {
+    private File createFile(String dicName) throws Exception {
         String timeStamp = new SimpleDateFormat("yyMMdd").format(Calendar.getInstance().getTime());
         String m_LogFileExt = ".log";
-        String logFilePath = m_MainPath + "\\Log\\" + timeStamp + m_LogFileExt;
+        String logFilePath = m_MainPath + "\\" + dicName + "\\" + timeStamp + m_LogFileExt;
         File logFile = new File(logFilePath);
         if (!logFile.exists()) {
             if(logFile.createNewFile()) {
Index: java/PTL/src/main/java/com/ptl/managers/AitReceiverSender.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/PTL/src/main/java/com/ptl/managers/AitReceiverSender.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/PTL/src/main/java/com/ptl/managers/AitReceiverSender.java	(date 1562916440695)
@@ -15,7 +15,7 @@
             this.dos = new DataOutputStream(s.getOutputStream());
         } catch (IOException e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
Index: java/HBM/src/main/java/com/hbm/daos/models/AitUserHostDAO.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/com/hbm/daos/models/AitUserHostDAO.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/com/hbm/daos/models/AitUserHostDAO.java	(date 1562918323396)
@@ -4,6 +4,7 @@
 import com.hbm.interfaces.AitUserHostDAOInterface;
 import com.hbm.models.AitUserHost;
 import com.hbm.models.entities.AitUserHostEntity;
+import com.ptl.managers.AitLogger;
 import org.hibernate.Session;
 import org.hibernate.query.Query;
 
@@ -18,7 +19,7 @@
 
     @Override
     public List<AitUserHost> findUserHostsByAccountId(int id) {
-        logger.info("opening: AitUserHostDAO.findUserHostsByAccountId(int)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserHostDAO.findUserHostsByAccountId(int)");
 
         Query query = getSession().createQuery("from usershosts where uhs_accid = :id", AitUserHostEntity.class);
         query.setParameter("id", id);
@@ -32,13 +33,13 @@
             }
         }
 
-        logger.info("exiting: AitUserHostDAO.findUserHostsByAccountId(int)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserHostDAO.findUserHostsByAccountId(int)");
         return result;
     }
 
     @Override
     public List<AitUserHost> findUserHostsByHostName(String hostName) {
-        logger.info("opening: AitUserHostDAO.findUserHostsByHostName(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserHostDAO.findUserHostsByHostName(String)");
 
         Query query = getSession().createQuery("from usershosts where uhs_hostname = :hostname", AitUserHostEntity.class);
         query.setParameter("hostname", hostName);
@@ -52,7 +53,7 @@
             }
         }
 
-        logger.info("exiting: AitUserHostDAO.findUserHostsByHostName(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserHostDAO.findUserHostsByHostName(String)");
         return result;
     }
 }
Index: java/PTL/src/main/java/com/ptl/managers/AitCrypter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/PTL/src/main/java/com/ptl/managers/AitCrypter.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/PTL/src/main/java/com/ptl/managers/AitCrypter.java	(date 1562916536885)
@@ -27,7 +27,7 @@
         }
         catch (NoSuchAlgorithmException e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
@@ -43,7 +43,7 @@
             return Base64.getEncoder().encodeToString(cipher.doFinal(strToEncrypt.getBytes(StandardCharsets.UTF_8)));
         } catch (Exception e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
@@ -60,7 +60,7 @@
             return new String(cipher.doFinal(Base64.getDecoder().decode(strToDecrypt)));
         } catch (Exception e) {
             try {
-                AitLogger.getInstance().logToFile(e.getStackTrace(), e.toString());
+                AitLogger.getInstance().logErrorToFile(e);
             } catch (Exception exc) {
                 e.printStackTrace();
                 exc.printStackTrace();
Index: java/GUI/src/main/java/com/gui/context/AitInitializer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/com/gui/context/AitInitializer.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/com/gui/context/AitInitializer.java	(date 1562918273968)
@@ -1,17 +1,12 @@
 package com.gui.context;
 
-import com.gui.AppGUI;
 import com.gui.managers.AitCultureManager;
 import com.gui.strings.AitPolishStrings;
-import com.hbm.daos.AitDAOFactory;
-import com.hbm.daos.models.AitAccountDAO;
-import com.hbm.models.AitAccount;
+import com.ptl.managers.AitLogger;
 import javafx.scene.control.Alert;
 import javafx.scene.control.ButtonType;
-import org.apache.log4j.Logger;
 
 public class AitInitializer {
-    private static Logger logger = Logger.getLogger(AitMainContext.class);
 
     private static AitInitializer ourInstance = new AitInitializer();
 
@@ -23,14 +18,14 @@
     }
 
     public void registerAppSettings() {
-        logger.info("opening: AitInitializer.registerAppSettings()");
+        AitLogger.getInstance().logInfoToFile("opening: AitInitializer.registerAppSettings()");
         AitCultureManager.getInstance().setLanguage(AitPolishStrings.locale);
 
-        logger.info("exiting: AitInitializer.registerAppSettings()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitInitializer.registerAppSettings()");
     }
 
     public boolean testDBConnection() {
-        logger.info("opening: AitInitializer.testDBConnection()");
+        AitLogger.getInstance().logInfoToFile("opening: AitInitializer.testDBConnection()");
         try {
             AitMainContext.getSession(true);
         } catch (Exception e){
@@ -38,7 +33,7 @@
             alert.show();
             return false;
         }
-        logger.info("exiting: AitInitializer.testDBConnection()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitInitializer.testDBConnection()");
         return true;
     }
 }
Index: java/GUI/src/main/java/com/gui/abstracts/AitGenericController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/com/gui/abstracts/AitGenericController.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/com/gui/abstracts/AitGenericController.java	(date 1562918274180)
@@ -1,10 +1,8 @@
 package com.gui.abstracts;
 
 import com.gui.interfaces.AitGenericControllerInterface;
-import org.apache.log4j.Logger;
 
 import java.io.Serializable;
 
 public abstract class AitGenericController<T, ID extends Serializable> implements AitGenericControllerInterface {
-    protected static Logger logger = Logger.getLogger(AitGenericController.class);
 }
Index: java/GUI/src/main/java/com/gui/context/AitMainContext.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/com/gui/context/AitMainContext.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/com/gui/context/AitMainContext.java	(date 1562918274079)
@@ -14,11 +14,10 @@
 import com.hbm.daos.AitDAOFactory;
 import com.hbm.hibernate.AitHibernateUtil;
 import com.hbm.models.AitAccount;
-import com.hbm.models.AitUserData;
 import com.hbm.models.AitUserHost;
 import com.hbm.models.entities.AitUserHostEntity;
+import com.ptl.managers.AitLogger;
 import javafx.util.Pair;
-import org.apache.log4j.Logger;
 import org.hibernate.Session;
 
 import java.net.InetAddress;
@@ -31,16 +30,15 @@
     private static AitAccount user;
     private static Session sessionObj;
     private static Map<String, Pair<AitNamespaceInterface, AitGenericControllerInterface>> frames;
-    private static Logger logger = Logger.getLogger(AitMainContext.class);
 
     public static void initFrames() {
-        logger.info("opening: AitIdManager.initFrames()");
+        AitLogger.getInstance().logInfoToFile("opening: AitIdManager.initFrames()");
         frames = new HashMap<>();
         frames.put(AitControllersNameConstStrings.ARNO_NAMESPACE, new Pair<>(new AitArnoNamespace(), new AitArnoController()));
         frames.put(AitControllersNameConstStrings.LOGIN_NAMESPACE, new Pair<>(new AitLoginNamespace(), new AitLoginController()));
         frames.put(AitControllersNameConstStrings.REGISTRATION_NAMESPACE, new Pair<>(new AitRegistrationNamespace(), new AitRegistraionController()));
         frames.put(AitControllersNameConstStrings.DASHBOARD_NAMESPACE, new Pair<>(new AitDashboardNamespace(), new AitDashboardController()));
-        logger.info("exiting: AitIdManager.initFrames()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitIdManager.initFrames()");
     }
 
     public static AitNamespaceInterface getNamespace(String key) {
@@ -94,7 +92,7 @@
 
                     userHost.saveOrUpdate();
                 } catch (Exception e) {
-                    logger.error("error: AitMainContext.setAccount(AitAccount, boolean)", e);
+                    AitLogger.getInstance().logErrorToFile("error: AitMainContext.setAccount(AitAccount, boolean)", e);
                 }
             } else {
                 for (AitUserHost host : userHosts) {
Index: java/HBM/src/main/java/com/hbm/daos/models/AitSettingsDAO.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/com/hbm/daos/models/AitSettingsDAO.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/com/hbm/daos/models/AitSettingsDAO.java	(date 1562918168960)
@@ -4,6 +4,7 @@
 import com.hbm.interfaces.AitSettingDAOInterface;
 import com.hbm.models.AitSetting;
 import com.hbm.models.entities.AitSettingEntity;
+import com.ptl.managers.AitLogger;
 import org.hibernate.Session;
 import org.hibernate.query.Query;
 
@@ -17,14 +18,14 @@
 
     @Override
     public AitSetting findSettingById(int id) {
-        logger.info("opening: AitAccountDAO.findAccountById(Integer)");
-        logger.info("exiting: AitAccountDAO.findAccountById(Integer)");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findAccountById(Integer)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountById(Integer)");
         return new AitSetting(getSession(), findById(id));
     }
 
     @Override
     public AitSetting findSettingByName(String name) {
-        logger.info("opening: AitAccountDAO.findAccountByPass(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findAccountByPass(String)");
 
         Query query = getSession().createQuery("from settings where set_name = :name", AitSettingEntity.class);
         query.setParameter("name", name);
@@ -32,11 +33,11 @@
         List<AitSettingEntity> accounts = query.list();
         if(accounts != null && accounts.size() > 0)
         {
-            logger.info("exiting: AitAccountDAO.findAccountByPass(String)");
+            AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountByPass(String)");
             return new AitSetting(getSession(), accounts.get(0));
         }
 
-        logger.info("exiting: AitAccountDAO.findAccountByPass(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountByPass(String)");
         return null;
     }
 }
Index: java/GUI/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/module-info.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/module-info.java	(date 1562918273898)
@@ -7,7 +7,6 @@
     requires org.hibernate.orm.core;
     requires HBM;
     requires PTL;
-    requires log4j;
     requires gson;
 
     opens com.gui.panels to javafx.fxml;
Index: java/HBM/src/main/java/com/hbm/hibernate/AitHibernateUtil.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/com/hbm/hibernate/AitHibernateUtil.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/com/hbm/hibernate/AitHibernateUtil.java	(date 1562918273988)
@@ -1,6 +1,6 @@
 package com.hbm.hibernate;
 
-import org.apache.log4j.Logger;
+import com.ptl.managers.AitLogger;
 import org.hibernate.SessionFactory;
 import org.hibernate.cfg.Configuration;
 import org.hibernate.cfg.Environment;
@@ -12,14 +12,13 @@
 
 public class AitHibernateUtil {
 
-    private static Logger logger = Logger.getLogger(AitHibernateUtil.class);
     private Map.Entry<String,String> account;
     private SessionFactory sessionFactory;
 
     private static AitHibernateUtil ourInstance = new AitHibernateUtil();
 
     private AitHibernateUtil() {
-        logger.info("opening: AitHibernateUtil.AitHibernateUtil()");
+        AitLogger.getInstance().logInfoToFile("opening: AitHibernateUtil.AitHibernateUtil()");
         try {
             String workStation = "pcppudwel";
             InetAddress addr = InetAddress.getLocalHost();
@@ -30,9 +29,9 @@
                     : new AbstractMap.SimpleEntry<>("admin", "admin");
 
         } catch (UnknownHostException e) {
-            logger.error("AitHibernateUtil.AitHibernateUtil()", e);
+            AitLogger.getInstance().logErrorToFile("AitHibernateUtil.AitHibernateUtil()", e);
         }
-        logger.info("exiting: AitHibernateUtil.AitHibernateUtil()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitHibernateUtil.AitHibernateUtil()");
     }
 
     public static AitHibernateUtil getInstance() {
@@ -40,7 +39,7 @@
     }
 
     public SessionFactory getSessionFactory() {
-        logger.info("opening: AitHibernateUtil.getSessionFactory()");
+        AitLogger.getInstance().logInfoToFile("opening: AitHibernateUtil.getSessionFactory()");
         if (sessionFactory == null) {
             try {
                 Configuration config = new Configuration();
@@ -51,10 +50,10 @@
 
                 sessionFactory = config.buildSessionFactory();
             } catch (Exception e) {
-                logger.error("AitHibernateUtil.getSessionFactory()", e);
+                AitLogger.getInstance().logErrorToFile("AitHibernateUtil.getSessionFactory()", e);
             }
         }
-        logger.info("exiting: AitHibernateUtil.getSessionFactory()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitHibernateUtil.getSessionFactory()");
         return sessionFactory;
     }
 }
\ No newline at end of file
Index: java/HBM/src/main/java/com/hbm/daos/AitDAOFactory.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/com/hbm/daos/AitDAOFactory.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/com/hbm/daos/AitDAOFactory.java	(date 1562918274215)
@@ -6,7 +6,7 @@
 import com.hbm.daos.models.AitUserDataDAO;
 import com.hbm.daos.models.AitUserHostDAO;
 import com.hbm.generics.AitGenericDAO;
-import org.apache.log4j.Logger;
+import com.ptl.managers.AitLogger;
 import org.hibernate.Session;
 
 import java.util.HashMap;
@@ -14,30 +14,29 @@
 
 public class AitDAOFactory {
 
-    private static Logger logger = Logger.getLogger(AitDAOFactory.class);
     private Session session;
     private Map<Class<?>, AitGenericDAO<?,?>> createdDAOImpl;
 
     public AitDAOFactory(Session session) {
-        logger.info("opening: AitDAOFactory.AitDAOFactory()");
+        AitLogger.getInstance().logInfoToFile("opening: AitDAOFactory.AitDAOFactory()");
         this.session = session;
         this.createdDAOImpl = new HashMap<>();
-        logger.info("exiting: AitDAOFactory.AitDAOFactory()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitDAOFactory.AitDAOFactory()");
     }
 
     @SuppressWarnings({ "rawtypes", "unchecked" })
     private AitGenericDAO instantiateDAO(Class daoClass) {
-        logger.info("opening: AitDAOFactory.instantiateDAO()");
+        AitLogger.getInstance().logInfoToFile("opening: AitDAOFactory.instantiateDAO()");
         try {
             AitGenericDAO result = createdDAOImpl.get(daoClass);
             if(result == null) {
                 result = (AitGenericDAO)daoClass.getConstructor(Session.class).newInstance(session);
                 createdDAOImpl.put(daoClass, result);
             }
-            logger.info("exiting: AitDAOFactory.instantiateDAO()");
+            AitLogger.getInstance().logInfoToFile("exiting: AitDAOFactory.instantiateDAO()");
             return result;
         } catch (Exception ex) {
-            logger.error("exiting: AitDAOFactory.instantiateDAO()", ex);
+            AitLogger.getInstance().logErrorToFile("exiting: AitDAOFactory.instantiateDAO()", ex);
             throw new RuntimeException("Can not instantiate DAO: " + daoClass, ex);
         }
     }
Index: java/HBM/HBM.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/HBM.iml	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/HBM.iml	(date 1562918294013)
@@ -34,7 +34,6 @@
     <orderEntry type="module" module-name="PTL" />
     <orderEntry type="library" name="Maven: com.google.code.gson:gson:2.8.5" level="project" />
     <orderEntry type="library" name="Maven: org.jsoup:jsoup:1.11.3" level="project" />
-    <orderEntry type="library" name="Maven: log4j:log4j:1.2.17" level="project" />
     <orderEntry type="library" name="Maven: org.apache.maven:maven-model:3.6.0" level="project" />
     <orderEntry type="library" name="Maven: org.codehaus.plexus:plexus-utils:3.1.0" level="project" />
   </component>
Index: java/HBM/src/main/java/com/hbm/generics/AitGenericDAO.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/com/hbm/generics/AitGenericDAO.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/com/hbm/generics/AitGenericDAO.java	(date 1562918273946)
@@ -2,7 +2,7 @@
 
 import com.hbm.hibernate.AitHibernateUtil;
 import com.hbm.interfaces.AitGenericDAOInterface;
-import org.apache.log4j.Logger;
+import com.ptl.managers.AitLogger;
 import org.hibernate.Session;
 
 import java.io.Serializable;
@@ -10,16 +10,15 @@
 
 public class AitGenericDAO<T extends Serializable, ID extends Serializable> implements AitGenericDAOInterface<T, ID> {
 
-    protected static Logger logger = Logger.getLogger(AitGenericDAO.class);
     private Class<T> persistentClass;
     private Session session;
 
     protected AitGenericDAO(Session session) {
-        logger.info("opening: AitGenericDAO.AitGenericDAO(Session)");
+        AitLogger.getInstance().logInfoToFile("opening: AitGenericDAO.AitGenericDAO(Session)");
         this.session = session;
         this.persistentClass = (Class<T>) ((ParameterizedType) getClass()
                 .getGenericSuperclass()).getActualTypeArguments()[0];
-        logger.info("exiting: AitGenericDAO.AitGenericDAO(Session)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitGenericDAO.AitGenericDAO(Session)");
     }
 
     protected Session getSession() {
Index: java/HBM/src/main/java/com/hbm/daos/models/AitUserDataDAO.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/com/hbm/daos/models/AitUserDataDAO.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/com/hbm/daos/models/AitUserDataDAO.java	(date 1562918313285)
@@ -4,6 +4,7 @@
 import com.hbm.interfaces.AitUserDataDAOInterface;
 import com.hbm.models.AitUserData;
 import com.hbm.models.entities.AitUserDataEntity;
+import com.ptl.managers.AitLogger;
 import org.hibernate.Session;
 import org.hibernate.query.Query;
 
@@ -17,15 +18,15 @@
 
     @Override
     public AitUserData findUserDataById(int id) {
-        logger.info("opening: AitUserDataDAO.findUserDataById(int)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserDataDAO.findUserDataById(int)");
 
-        logger.info("exiting: AitUserDataDAO.findUserDataById(int)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserDataDAO.findUserDataById(int)");
         return new AitUserData(getSession(), findById(id));
     }
 
     @Override
     public List<AitUserData> findUserDataByNick(String nick) {
-        logger.info("opening: AitUserDataDAO.findUserDataByNick(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserDataDAO.findUserDataByNick(String)");
 
         Query query = getSession().createQuery("from usersdata where udt_nick = :nick", AitUserDataEntity.class);
         query.setParameter("nick", nick);
@@ -39,13 +40,13 @@
             }
         }
 
-        logger.info("exiting: AitUserDataDAO.findUserDataByNick(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserDataDAO.findUserDataByNick(String)");
         return result;
     }
 
     @Override
     public List<AitUserData> findUserDataByFirstName(String firstName) {
-        logger.info("opening: AitUserDataDAO.findUserDataByFirstName(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserDataDAO.findUserDataByFirstName(String)");
 
         Query query = getSession().createQuery("from usersdata where udt_firstname = :firstName", AitUserDataEntity.class);
         query.setParameter("firstName", firstName);
@@ -59,13 +60,13 @@
             }
         }
 
-        logger.info("exiting: AitUserDataDAO.findUserDataByFirstName(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserDataDAO.findUserDataByFirstName(String)");
         return result;
     }
 
     @Override
     public List<AitUserData> findUserDataByMiddleName(String middleName) {
-        logger.info("opening: AitUserDataDAO.findUserDataByMiddleName(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserDataDAO.findUserDataByMiddleName(String)");
 
         Query query = getSession().createQuery("from usersdata where udt_middlename = :middleName", AitUserDataEntity.class);
         query.setParameter("middleName", middleName);
@@ -79,13 +80,13 @@
             }
         }
 
-        logger.info("exiting: AitUserDataDAO.findUserDataByMiddleName(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserDataDAO.findUserDataByMiddleName(String)");
         return result;
     }
 
     @Override
     public List<AitUserData> findUserDataByLastName(String lastName) {
-        logger.info("opening: AitUserDataDAO.findUserDataByLastName(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserDataDAO.findUserDataByLastName(String)");
 
         Query query = getSession().createQuery("from usersdata where udt_lastname = :lastName", AitUserDataEntity.class);
         query.setParameter("lastName", lastName);
@@ -99,13 +100,13 @@
             }
         }
 
-        logger.info("exiting: AitUserDataDAO.findUserDataByLastName(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserDataDAO.findUserDataByLastName(String)");
         return result;
     }
 
     @Override
     public List<AitUserData> findAllUserData() {
-        logger.info("opening: AitUserDataDAO.findAllUserData()");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserDataDAO.findAllUserData()");
 
         Query query = getSession().createQuery("from usersdata", AitUserDataEntity.class);
 
@@ -118,13 +119,13 @@
             }
         }
 
-        logger.info("exiting: AitUserDataDAO.findAllUserData()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserDataDAO.findAllUserData()");
         return result;
     }
 
     @Override
     public AitUserData findUserDataByAccountId(int id) {
-        logger.info("opening: AitUserDataDAO.findUserHostsByAccountId(int)");
+        AitLogger.getInstance().logInfoToFile("opening: AitUserDataDAO.findUserHostsByAccountId(int)");
 
         Query query = getSession().createQuery("from usersdata where udt_accid = :id", AitUserDataEntity.class);
         query.setParameter("id", id);
@@ -138,7 +139,7 @@
             }
         }
 
-        logger.info("exiting: AitUserDataDAO.findUserHostsByAccountId(int)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitUserDataDAO.findUserHostsByAccountId(int)");
         return result.isEmpty() ?  null : result.get(0);
     }
 }
Index: java/HBM/src/main/java/com/hbm/daos/models/AitAccountDAO.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/com/hbm/daos/models/AitAccountDAO.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/com/hbm/daos/models/AitAccountDAO.java	(date 1562918302662)
@@ -4,6 +4,7 @@
 import com.hbm.interfaces.AitAccountDAOInterface;
 import com.hbm.models.AitAccount;
 import com.hbm.models.entities.AitAccountEntity;
+import com.ptl.managers.AitLogger;
 import org.hibernate.Session;
 import org.hibernate.query.Query;
 
@@ -17,14 +18,14 @@
 
     @Override
     public AitAccount findAccountById(int id) {
-        logger.info("opening: AitAccountDAO.findAccountById(Integer)");
-        logger.info("exiting: AitAccountDAO.findAccountById(Integer)");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findAccountById(Integer)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountById(Integer)");
         return new AitAccount(getSession(), findById(id));
     }
 
     @Override
     public List<AitAccount> findAccountByLogin(String login) {
-        logger.info("opening: AitAccountDAO.findAccountByLogin(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findAccountByLogin(String)");
 
         Query query = getSession().createQuery("from accounts where acc_login = :login", AitAccountEntity.class);
         query.setParameter("login", login);
@@ -38,13 +39,13 @@
             }
         }
 
-        logger.info("exiting: AitAccountDAO.findAccountByLogin(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountByLogin(String)");
         return result;
     }
 
     @Override
     public AitAccount findAccountByPass(String pass) {
-        logger.info("opening: AitAccountDAO.findAccountByPass(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findAccountByPass(String)");
 
         Query query = getSession().createQuery("from accounts where acc_pass = :pass", AitAccountEntity.class);
         query.setParameter("pass", pass);
@@ -52,17 +53,17 @@
         List<AitAccountEntity> accounts = query.list();
         if(accounts != null && accounts.size() > 0)
         {
-            logger.info("exiting: AitAccountDAO.findAccountByPass(String)");
+            AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountByPass(String)");
             return new AitAccount(getSession(), accounts.get(0));
         }
 
-        logger.info("exiting: AitAccountDAO.findAccountByPass(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountByPass(String)");
         return null;
     }
 
     @Override
     public List<AitAccount> findAccountByEmail(String email) {
-        logger.info("opening: AitAccountDAO.findAccountByEmail(String)");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findAccountByEmail(String)");
 
         Query query = getSession().createQuery("from accounts where acc_email = :email", AitAccountEntity.class);
         query.setParameter("email", email);
@@ -76,13 +77,13 @@
             }
         }
 
-        logger.info("exiting: AitAccountDAO.findAccountByEmail(String)");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAccountByEmail(String)");
         return result;
     }
 
     @Override
     public List<AitAccount> findActiveAccounts() {
-        logger.info("opening: AitAccountDAO.findActiveAccounts()");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findActiveAccounts()");
 
         Query query = getSession().createQuery("from accounts where acc_active = 1", AitAccountEntity.class);
 
@@ -95,13 +96,13 @@
             }
         }
 
-        logger.info("exiting: AitAccountDAO.findActiveAccounts()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findActiveAccounts()");
         return result;
     }
 
     @Override
     public List<AitAccount> findNotActiveAccounts() {
-        logger.info("opening: AitAccountDAO.findNotActiveAccounts()");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findNotActiveAccounts()");
 
         Query query = getSession().createQuery("from accounts where acc_active = 0", AitAccountEntity.class);
 
@@ -114,13 +115,13 @@
             }
         }
 
-        logger.info("exiting: AitAccountDAO.findNotActiveAccounts()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findNotActiveAccounts()");
         return result;
     }
 
     @Override
     public List<AitAccount> findAllAccount() {
-        logger.info("opening: AitAccountDAO.findAllAccount()");
+        AitLogger.getInstance().logInfoToFile("opening: AitAccountDAO.findAllAccount()");
 
         Query query = getSession().createQuery("from accounts", AitAccountEntity.class);
 
@@ -133,7 +134,7 @@
             }
         }
 
-        logger.info("exiting: AitAccountDAO.findAllAccount()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitAccountDAO.findAllAccount()");
         return result;
     }
 }
Index: java/GUI/src/main/java/com/gui/panels/AitDashboardController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/com/gui/panels/AitDashboardController.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/com/gui/panels/AitDashboardController.java	(date 1562918274095)
@@ -7,9 +7,8 @@
 import com.gui.managers.AitCultureManager;
 import com.gui.strings.AitControllersNameConstStrings;
 import com.gui.strings.AitFramesStrings;
-import com.hbm.daos.AitDAOFactory;
 import com.hbm.models.AitAccount;
-import com.hbm.models.AitUserData;
+import com.ptl.managers.AitLogger;
 import javafx.fxml.FXML;
 import javafx.scene.Scene;
 import javafx.scene.control.Alert;
@@ -40,7 +39,7 @@
 
     @FXML
     public void initialize() {
-        logger.info("opening: AitDashboardController.initialize()");
+        AitLogger.getInstance().logInfoToFile("opening: AitDashboardController.initialize()");
 
         fileMenu.setText(AitCultureManager.getInstance().getLanguage().getFileMenuContent());
         editMenu.setText(AitCultureManager.getInstance().getLanguage().getEditMenuContent());
@@ -83,14 +82,14 @@
                 AitMainContext.setNamespaceTitle(AitControllersNameConstStrings.DASHBOARD_NAMESPACE, account.getUserData().getNick());
             }
         } catch (UnknownHostException e) {
-            logger.error("error: AitDashboardController.initialize()", e);
+            AitLogger.getInstance().logErrorToFile("error: AitDashboardController.initialize()", e);
         }
 
-        logger.info("exiting: AitDashboardController.initialize()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitDashboardController.initialize()");
     }
 
     private void onLogoutAction() {
-        logger.info("opening: AitDashboardController.onLogoutAction()");
+        AitLogger.getInstance().logInfoToFile("opening: AitDashboardController.onLogoutAction()");
         // TODO show progress bar
 
         try {
@@ -102,9 +101,9 @@
                 AppGUI.setRoot(AitControllersNameConstStrings.LOGIN_NAMESPACE, AitControllersNameConstStrings.DASHBOARD_NAMESPACE, this);
             }
         } catch (Exception e) {
-            logger.error("error: AitDashboardController.onLogoutAction()");
+            AitLogger.getInstance().logErrorToFile("error: AitDashboardController.onLogoutAction()", e);
         }
-        logger.info("exiting: AitDashboardController.onLogoutAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitDashboardController.onLogoutAction()");
     }
 
     private void onExitAction() {
@@ -120,7 +119,7 @@
     }
 
     private void onDoshboardAction() {
-        logger.info("exiting: AitDashboardController.onCreateAccountAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitDashboardController.onCreateAccountAction()");
         Alert dialog = new Alert(Alert.AlertType.NONE, "Do you want open new dashboard?", ButtonType.YES, ButtonType.NO);
         dialog.showAndWait();
         if(dialog.getResult() == ButtonType.YES) {
@@ -137,14 +136,14 @@
                 stage.show();
                 AppGUI.addStage(stage);
             } catch (Exception e) {
-                logger.error("error: AitDashboardController.onCreateAccountAction()",e);
+                AitLogger.getInstance().logErrorToFile("error: AitDashboardController.onCreateAccountAction()",e);
             }
         }
-        logger.info("exiting: AitDashboardController.onLogoutAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitDashboardController.onLogoutAction()");
     }
 
     private void onCreateAccountAction() {
-        logger.info("exiting: AitDashboardController.onCreateAccountAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitDashboardController.onCreateAccountAction()");
         try {
             Stage stage = new Stage();
             AitNamespaceInterface namespace = AitMainContext.getNamespace(AitControllersNameConstStrings.DASHBOARD_NAMESPACE);
@@ -158,8 +157,8 @@
             stage.show();
             AppGUI.addStage(stage);
         } catch (Exception e) {
-            logger.error("error: AitDashboardController.onCreateAccountAction()",e);
+            AitLogger.getInstance().logErrorToFile("error: AitDashboardController.onCreateAccountAction()",e);
         }
-        logger.info("exiting: AitDashboardController.onLogoutAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitDashboardController.onLogoutAction()");
     }
 }
Index: java/HBM/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/src/main/java/module-info.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/src/main/java/module-info.java	(date 1562918274147)
@@ -9,7 +9,6 @@
     requires com.sun.xml.bind;
     requires org.hibernate.orm.core;
     requires mysql.connector.java;
-    requires log4j;
     requires PTL;
     requires maven.model;
 
Index: java/GUI/src/main/java/com/gui/panels/AitArnoController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/com/gui/panels/AitArnoController.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/com/gui/panels/AitArnoController.java	(date 1562918168819)
@@ -8,6 +8,7 @@
 import com.gui.managers.AitCultureManager;
 import com.gui.strings.AitControllersNameConstStrings;
 import com.hbm.models.AitAccount;
+import com.ptl.managers.AitLogger;
 import javafx.concurrent.Task;
 import javafx.fxml.FXML;
 import javafx.scene.control.Alert;
@@ -28,7 +29,7 @@
 
     @FXML
     public void initialize() {
-        logger.info("opening: AitArnoController.initialize()");
+        AitLogger.getInstance().logInfoToFile("opening: AitArnoController.initialize()");
         BindableTask task = new BindableTask(this);
         progressBar.progressProperty().bind(task.progressProperty());
         arnoFirstInfo.textProperty().bind(task.messageProperty());
@@ -39,9 +40,9 @@
         try {
             arnoSecondInfo.setText(AitCultureManager.getInstance().getLanguage().getSecondInfoProgress());
         } catch (Exception e) {
-            logger.error("error: AitArnoController.initialize()", e);
+            AitLogger.getInstance().logErrorToFile("error: AitArnoController.initialize()", e);
         }
-        logger.info("exiting: AitArnoController.initialize()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitArnoController.initialize()");
     }
 
     private class BindableTask extends Task<Integer> {
@@ -71,7 +72,7 @@
                 updateProgress(10,10);
                 Thread.sleep(200);
             } catch (SecurityException se) {
-                logger.error("eror: ArnoControllerBandableTask.succeeded()", se);
+                AitLogger.getInstance().logErrorToFile("eror: ArnoControllerBandableTask.succeeded()", se);
                 Alert alert = new Alert(Alert.AlertType.ERROR, "You dont have permition to use this app! Run app as administator!", ButtonType.OK);
                 alert.show();
             } catch (Exception e) {
Index: java/GUI/GUI.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/GUI.iml	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/GUI.iml	(date 1562918294034)
@@ -43,7 +43,6 @@
     <orderEntry type="module" module-name="PTL" />
     <orderEntry type="library" name="Maven: com.google.code.gson:gson:2.8.5" level="project" />
     <orderEntry type="library" name="Maven: org.jsoup:jsoup:1.11.3" level="project" />
-    <orderEntry type="library" name="Maven: log4j:log4j:1.2.17" level="project" />
     <orderEntry type="library" name="Maven: org.apache.maven:maven-model:3.6.0" level="project" />
     <orderEntry type="library" name="Maven: org.codehaus.plexus:plexus-utils:3.1.0" level="project" />
   </component>
Index: java/HBM/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/HBM/pom.xml	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/HBM/pom.xml	(date 1562918255817)
@@ -48,11 +48,6 @@
             <version>1.0.0</version>
             <scope>compile</scope>
         </dependency>
-        <dependency>
-            <groupId>log4j</groupId>
-            <artifactId>log4j</artifactId>
-            <version>1.2.17</version>
-        </dependency>
     </dependencies>
 
 </project>
\ No newline at end of file
Index: java/GUI/src/main/java/com/gui/panels/AitRegistraionController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/com/gui/panels/AitRegistraionController.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/com/gui/panels/AitRegistraionController.java	(date 1562918168930)
@@ -12,6 +12,7 @@
 import com.hbm.models.entities.AitAccountEntity;
 import com.hbm.models.entities.AitUserDataEntity;
 import com.ptl.managers.AitCrypter;
+import com.ptl.managers.AitLogger;
 import com.ptl.managers.AitMailSender;
 import javafx.fxml.FXML;
 import javafx.scene.control.Alert;
@@ -60,7 +61,7 @@
 
     @FXML
     public void initialize() {
-        logger.info("opening: AitRegistraionController.initialize()");
+        AitLogger.getInstance().logInfoToFile("opening: AitRegistraionController.initialize()");
         try {
             reqLabel.setText(AitCultureManager.getInstance().getLanguage().getReqLabelContent());
             backButton.setText(AitCultureManager.getInstance().getLanguage().getBackButtonContent());
@@ -90,14 +91,14 @@
             }
 
         } catch (Exception e) {
-            logger.error("error: RegistrationController.initialize()", e);
+            AitLogger.getInstance().logErrorToFile("error: RegistrationController.initialize()", e);
         }
-        logger.info("exiting: AitRegistraionController.initialize()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitRegistraionController.initialize()");
     }
 
     @FXML
     public void onRegisterAction() {
-        logger.info("opening: AitRegistraionController.onRegisterAction()");
+        AitLogger.getInstance().logInfoToFile("opening: AitRegistraionController.onRegisterAction()");
         try {
             AitMainContext.getSession(true).beginTransaction();
             // TODO show progress bar
@@ -172,14 +173,14 @@
                         alert.show();
                         nickBox.requestFocus();
                         nickBox.clear();
-                        logger.info("exiting: AitRegistraionController.onRegisterAction()");
+                        AitLogger.getInstance().logInfoToFile("exiting: AitRegistraionController.onRegisterAction()");
                         return;
                     } else if (!datas.isEmpty()) {
                         Alert alert = new Alert(Alert.AlertType.WARNING, "Nick must be unique!", ButtonType.OK);
                         alert.show();
                         nickBox.requestFocus();
                         nickBox.clear();
-                        logger.info("exiting: AitRegistraionController.onRegisterAction()");
+                        AitLogger.getInstance().logInfoToFile("exiting: AitRegistraionController.onRegisterAction()");
                         return;
                     }
                 }
@@ -190,7 +191,7 @@
                     alert.show();
                     middleNameBox.requestFocus();
                     middleNameBox.clear();
-                    logger.info("exiting: AitRegistraionController.onRegisterAction()");
+                    AitLogger.getInstance().logInfoToFile("exiting: AitRegistraionController.onRegisterAction()");
                     return;
                 }
 
@@ -206,7 +207,7 @@
                         alert.show();
                         birthdateBox.requestFocus();
                         birthdateBox.setValue(null);
-                        logger.info("exiting: AitRegistraionController.onRegisterAction()");
+                        AitLogger.getInstance().logInfoToFile("exiting: AitRegistraionController.onRegisterAction()");
                         return;
                     }
                 }
@@ -248,16 +249,16 @@
                 }
             }
         } catch (Exception e) {
-            logger.error("error: AitRegistraionController.onRegisterAction()", e);
+            AitLogger.getInstance().logErrorToFile("error: AitRegistraionController.onRegisterAction()", e);
         }
 
-        logger.info("exiting: AitRegistraionController.onRegisterAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitRegistraionController.onRegisterAction()");
     }
 
     public void onBackAction() {
-        logger.info("opening: AitRegistraionController.onRegisterAction()");
+        AitLogger.getInstance().logInfoToFile("opening: AitRegistraionController.onRegisterAction()");
         AppGUI.back();
-        logger.info("exiting: AitRegistraionController.onRegisterAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitRegistraionController.onRegisterAction()");
     }
 
 }
\ No newline at end of file
Index: java/GUI/src/main/java/com/gui/panels/AitLoginController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- java/GUI/src/main/java/com/gui/panels/AitLoginController.java	(revision ba94f2d9bbad5818b9bd5c27a00e5bfb62c952cd)
+++ java/GUI/src/main/java/com/gui/panels/AitLoginController.java	(date 1562918169034)
@@ -8,6 +8,7 @@
 import com.hbm.daos.AitDAOFactory;
 import com.hbm.models.AitAccount;
 import com.ptl.managers.AitCrypter;
+import com.ptl.managers.AitLogger;
 import javafx.fxml.FXML;
 import javafx.scene.control.Alert;
 import javafx.scene.control.Button;
@@ -37,7 +38,7 @@
 
     @FXML
     public void initialize() {
-        logger.info("opening: AitLoginController.initialize()");
+        AitLogger.getInstance().logInfoToFile("opening: AitLoginController.initialize()");
         try {
             loginBox.setPromptText(AitCultureManager.getInstance().getLanguage().getLoginPrompt());
             passwordBox.setPromptText(AitCultureManager.getInstance().getLanguage().getPasswordPrompt());
@@ -51,13 +52,13 @@
                 try {
                     openRegisterFrame();
                 } catch (Exception e) {
-                    logger.error("error: AitLoginController.setOnAction()", e);
+                    AitLogger.getInstance().logErrorToFile("error: AitLoginController.setOnAction()", e);
                 }
             });
         } catch (Exception e) {
-            logger.error("error: AitLoginController.initialize()", e);
+            AitLogger.getInstance().logErrorToFile("error: AitLoginController.initialize()", e);
         }
-        logger.info("exiting: AitLoginController.initialize()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitLoginController.initialize()");
     }
 
     private void onKeyReleased(KeyEvent keyEvent) {
@@ -68,14 +69,14 @@
 
     @FXML
     private void openRegisterFrame() throws Exception {
-        logger.info("opening: AitLoginController.openRegisterFrame()");
+        AitLogger.getInstance().logInfoToFile("opening: AitLoginController.openRegisterFrame()");
         AppGUI.setRoot(AitControllersNameConstStrings.REGISTRATION_NAMESPACE, AitControllersNameConstStrings.LOGIN_NAMESPACE, this);
-        logger.info("exiting: AitLoginController.openRegisterFrame()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitLoginController.openRegisterFrame()");
     }
 
     @FXML
     private void loginAction() {
-        logger.info("opening: AitLoginController.loginAction()");
+        AitLogger.getInstance().logInfoToFile("opening: AitLoginController.loginAction()");
         try {
             // TODO show progress bar
 
@@ -86,13 +87,13 @@
             if(login == null || login.length() == 0) {
                 Alert alert = new Alert(Alert.AlertType.ERROR, "Please enter login!", ButtonType.OK);
                 alert.show();
-                logger.info("Empty login!");
+                AitLogger.getInstance().logInfoToFile("Empty login!");
                 loginBox.requestFocus();
             }
             else if(password == null || password.length() == 0) {
                 Alert alert = new Alert(Alert.AlertType.ERROR, "Please enter password!", ButtonType.OK);
                 alert.show();
-                logger.info("Empty password!");
+                AitLogger.getInstance().logInfoToFile("Empty password!");
                 passwordBox.requestFocus();
             }
             else {
@@ -104,15 +105,16 @@
                             if(!acc.IsActive()) {
                                 Alert alert = new Alert(Alert.AlertType.ERROR, "AitAccount is not activate!", ButtonType.OK);
                                 alert.show();
-                                logger.info("AitAccount is not activate!");
+                                AitLogger.getInstance().logInfoToFile("AitAccount is not activate!");
                                 passwordBox.requestFocus();
-                                logger.info("exiting: AitLoginController.loginAction()");
+                                AitLogger.getInstance().logInfoToFile("exiting: AitLoginController.loginAction()");
                                 return;
                             }
                             else {
                                 try {
                                     AitMainContext.setAccount(acc, rememberMe);
                                 } catch (Exception se) {
+                                    AitLogger.getInstance().logErrorToFile("",se);
                                     Alert alert = new Alert(Alert.AlertType.ERROR, "You dont have permition to use this app! Run app as administator!", ButtonType.OK);
                                     alert.showAndWait();
                                     if(alert.getResult() == ButtonType.OK) {
@@ -121,28 +123,28 @@
                                 }
                                 AppGUI.setRoot(AitControllersNameConstStrings.DASHBOARD_NAMESPACE, AitControllersNameConstStrings.LOGIN_NAMESPACE, this, true);
 
-                                logger.info("exiting: AitLoginController.loginAction() Login Successfull");
+                                AitLogger.getInstance().logInfoToFile("exiting: AitLoginController.loginAction() Login Successfull");
                                 return;
                             }
                         }
                     }
                     Alert alert = new Alert(Alert.AlertType.ERROR, "Incorect password!", ButtonType.OK);
                     alert.show();
-                    logger.info("Incorect password!");
+                    AitLogger.getInstance().logInfoToFile("Incorect password!");
                     passwordBox.requestFocus();
-                    logger.info("exiting: AitLoginController.loginAction()");
+                    AitLogger.getInstance().logInfoToFile("exiting: AitLoginController.loginAction()");
                     return;
                 }
                 else {
                     Alert alert = new Alert(Alert.AlertType.ERROR, "Login not found!", ButtonType.OK);
                     alert.show();
-                    logger.info("Login not found!");
+                    AitLogger.getInstance().logInfoToFile("Login not found!");
                     loginBox.requestFocus();
                 }
             }
         } catch (Exception e) {
-            logger.error("error: AitLoginController.loginAction()", e);
+            AitLogger.getInstance().logErrorToFile("error: AitLoginController.loginAction()", e);
         }
-        logger.info("exiting: AitLoginController.loginAction()");
+        AitLogger.getInstance().logInfoToFile("exiting: AitLoginController.loginAction()");
     }
 }
