Index: csharp/WPF/Interfaces/Triggerable.cs
===================================================================
--- csharp/WPF/Interfaces/Triggerable.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Interfaces/Triggerable.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
@@ -1,9 +0,0 @@
-namespace WPF.Interfaces
-{
-    public interface Triggerable
-    {
-        void Add();
-        void Update();
-        void Remove();
-    }
-}
Index: csharp/WPF/Databases/Models/AitFilesModel.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Databases/Models/AitFilesModel.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Databases/Models/AitFilesModel.cs	(date 1564040799363)
@@ -3,6 +3,7 @@
 using System.ComponentModel.DataAnnotations.Schema;
 using System.Runtime.Serialization;
 using WPF.Databases.Contexts;
+using WPF.Enums;
 using WPF.ExtendedClasses;
 using WPF.Validators;
 
@@ -21,7 +22,7 @@
             get { return BaseID; }
             set
             {
-                if (BasePropertiesValidator.ValidateID(value))
+                if (BasePropertiesValidator.ValidatePrimaryKey(value, TablePrefix))
                     SetField(ref BaseID, value, nameof(ID));
             }
         }
@@ -81,6 +82,9 @@
             set { SetField(ref BaseLastUpdate, value, nameof(LastUpdate)); }
         }
 
+        [NotMapped]
+        public IDInerfixEnum TablePrefix { get { return IDInerfixEnum.FLS; } }
+
         public AitAccountModel FileCreator { get; set; }
         public AitAccountModel FileOwner
         {
Index: csharp/WPF/Databases/Models/AitAccountModel.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Databases/Models/AitAccountModel.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Databases/Models/AitAccountModel.cs	(date 1564040788057)
@@ -22,7 +22,7 @@
             get { return BaseID; }
             set
             {
-                if (BasePropertiesValidator.ValidateID(value))
+                if (BasePropertiesValidator.ValidatePrimaryKey(value, TablePrefix))
                     SetField(ref BaseID, value, nameof(ID));
             }
         }
@@ -88,6 +88,9 @@
             set { SetField(ref BaseLastUpdate, value, nameof(LastUpdate)); }
         }
 
+        [NotMapped]
+        public IDInerfixEnum TablePrefix { get { return IDInerfixEnum.ACC; } }
+
         public AitAccountModel(DBContext context) : base(context)
         {
             IsActive = false;
Index: csharp/UTW/Tests/DBActionsUnitTests.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/UTW/Tests/DBActionsUnitTests.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/UTW/Tests/DBActionsUnitTests.cs	(date 1564040270712)
@@ -4,20 +4,23 @@
 using System.Linq;
 using WPF.Databases.Contexts;
 using WPF.Databases.Models;
+using WPF.Enums;
+using WPF.Managers.Helpers;
 
 namespace UTW.Tests
 {
     [TestClass]
     public class DBActionsUnitTests
     {
+        private static readonly string id = Generator.IDGenerator(IDInerfixEnum.ACC);
+
         [TestMethod]
         public void InsertTest()
         {
             using (var dataContext = PDBContext.Instance.Context)
             {
-                dataContext.ReCreate();
+                //dataContext.ReCreate();
 
-                var id = "AIT-ACC-0000001";
                 if (!dataContext.Accounts.Any(q => q.ID.Equals(id)) && !dataContext.Stsgenids.Any(q => q.ID.Equals(id)))
                 {
                     var acc = new AitAccountModel(dataContext)
@@ -25,7 +28,7 @@
                         ID = id,
                         Login = "test"
                     };
-                    acc.Add();
+                    acc.Insert();
                     acc.Context.SaveChanges();
                 }
                 Assert.IsTrue(dataContext.Accounts.Any(q => q.ID.Equals(id)));
@@ -46,13 +49,13 @@
         {
             using (var dbContext = PDBContext.Instance.Context)
             {
-                var acc = dbContext.Accounts.Where(q => q.ID.Equals("AIT-ACC-0000001")).FirstOrDefault();
+                var acc = dbContext.Accounts.Where(q => q.ID.Equals(id)).FirstOrDefault();
                 if(acc != null)
                 {
-                    acc.IsActive = false;
+                    acc.IsActive = true;
                     acc.Update();
                     dbContext.SaveChanges();
-                    Assert.IsFalse(dbContext.Accounts.Where(q => q.ID.Equals("AIT-ACC-0000001")).FirstOrDefault().IsActive);
+                    Assert.IsTrue(dbContext.Accounts.Where(q => q.ID.Equals(id)).FirstOrDefault().IsActive);
                 }
             }
         }
@@ -62,15 +65,17 @@
         {
             using (var dbContext = PDBContext.Instance.Context)
             {
-                var acc = dbContext.Accounts.Where(q => q.ID.Equals("AIT-ACC-0000001")).FirstOrDefault();
+                var acc = dbContext.Accounts.Where(q => q.ID.Equals(id)).FirstOrDefault();
                 if(acc != null)
                 {
-                    dbContext.Remove(acc);
+                    acc.Delete();
                 }
-                var sts = dbContext.Stsgenids.Where(q => q.ID.Equals("AIT-ACC-0000001")).FirstOrDefault();
+                var sts = dbContext.Stsgenids.Where(q => q.ID.Equals(id)).FirstOrDefault();
 
                 Assert.IsNotNull(sts);
                 Assert.IsTrue(sts.Delete != null);
+
+                dbContext.SaveChanges();
             }
         }
     }
Index: csharp/WPF/Databases/Models/AitUserHostModel.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Databases/Models/AitUserHostModel.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Databases/Models/AitUserHostModel.cs	(date 1564040799354)
@@ -3,6 +3,7 @@
 using System.ComponentModel.DataAnnotations.Schema;
 using System.Runtime.Serialization;
 using WPF.Databases.Contexts;
+using WPF.Enums;
 using WPF.Validators;
 
 namespace WPF.Databases.Models
@@ -18,7 +19,7 @@
             get { return BaseID; }
             set
             {
-                if (BasePropertiesValidator.ValidateID(value))
+                if (BasePropertiesValidator.ValidatePrimaryKey(value, TablePrefix))
                     SetField(ref BaseID, value, nameof(ID));
             }
         }
@@ -47,6 +48,9 @@
             set { SetField(ref BaseLastUpdate, value, nameof(LastUpdate)); }
         }
 
+        [NotMapped]
+        public IDInerfixEnum TablePrefix { get { return IDInerfixEnum.USH; } }
+
         public AitAccountModel AccountData { get; set; }
 
         public AitUserHostModel(DBContext context) : base(context)
Index: csharp/WPF/Databases/Models/AitUserDataModel.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Databases/Models/AitUserDataModel.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Databases/Models/AitUserDataModel.cs	(date 1564040799358)
@@ -2,6 +2,7 @@
 using System.ComponentModel.DataAnnotations;
 using System.ComponentModel.DataAnnotations.Schema;
 using System.Runtime.Serialization;
+using WPF.Enums;
 using WPF.Validators;
 
 namespace WPF.Databases.Models
@@ -18,7 +19,7 @@
             get { return BaseID; }
             set
             {
-                if (BasePropertiesValidator.ValidateID(value))
+                if (BasePropertiesValidator.ValidatePrimaryKey(value, TablePrefix))
                     SetField(ref BaseID, value, nameof(ID));
             }
         }
@@ -94,6 +95,9 @@
             }
         }
 
+        [NotMapped]
+        public IDInerfixEnum TablePrefix { get { return IDInerfixEnum.USD; } }
+
         public AitAccountModel AccountData { get; set; }
 
         public AitUserDataModel() : base(null)
Index: csharp/WPF/WPF.csproj
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/WPF.csproj	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/WPF.csproj	(date 1564040061753)
@@ -202,7 +202,9 @@
     <Compile Include="Databases\Contexts\PDBContext.cs" />
     <Compile Include="Databases\Models\BaseEntityModel.cs" />
     <Compile Include="Enums\FileTypeEnum.cs" />
+    <Compile Include="Enums\IDPrefixEnum.cs" />
     <Compile Include="Exceptions\BaseExceptions.cs" />
+    <Compile Include="Exceptions\GenerateExceptions.cs" />
     <Compile Include="Exceptions\SqliteExceptions.cs" />
     <Compile Include="Extensions\NotifyPropertyChangedExtension.cs" />
     <Compile Include="Databases\Models\AitFilesModel.cs" />
@@ -210,7 +212,7 @@
     <Compile Include="Databases\Models\AitUserHostModel.cs" />
     <Compile Include="Databases\Models\SysStsgenids.cs" />
     <Compile Include="Interfaces\Initializable.cs" />
-    <Compile Include="Interfaces\Triggerable.cs" />
+    <Compile Include="Interfaces\ITriggerable.cs" />
     <Compile Include="Managers\CryptoJsonManager.cs" />
     <Compile Include="Managers\EncryptionManager.cs" />
     <Compile Include="Managers\ExceptionManager.cs" />
Index: csharp/UTW/UTW.csproj
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/UTW/UTW.csproj	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/UTW/UTW.csproj	(date 1564037787076)
@@ -156,6 +156,7 @@
     <Compile Include="Tests\DBActionsUnitTests.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="Tests\FileActionUnitTest.cs" />
+    <Compile Include="Tests\GeneratorActionUnitTest.cs" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\WPF\WPF.csproj">
Index: csharp/WPF/Databases/Models/SysStsgenids.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Databases/Models/SysStsgenids.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Databases/Models/SysStsgenids.cs	(date 1564040934096)
@@ -1,6 +1,7 @@
 using System;
 using System.ComponentModel.DataAnnotations;
 using System.ComponentModel.DataAnnotations.Schema;
+using WPF.Enums;
 using WPF.Validators;
 
 namespace WPF.Databases.Models
Index: csharp/WPF/Managers/Helpers/Generator.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Managers/Helpers/Generator.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Managers/Helpers/Generator.cs	(date 1564041047770)
@@ -1,37 +1,85 @@
 using System;
+using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
+using WPF.Databases.Contexts;
+using WPF.Databases.Models;
+using WPF.Enums;
+using WPF.Exceptions;
+using WPF.Validators;
 
 namespace WPF.Managers.Helpers
 {
     public class Generator
     {
-        public static string GenerateHashID()
+        private static readonly char separator = '-';
+
+        public static string IDGenerator(IDInerfixEnum tablePrefix)
         {
-            return DateTime.Now.ToUniversalTime().GetHashCode().ToString();
-        }
+            return IDGenerator(IDPrefixEnum.AIT, tablePrefix);
+        }
+
+        public static string IDGenerator(IDPrefixEnum prefix, IDInerfixEnum inerfix)
+        {
+            var id = "0000000";
+
+            using(var context = PDBContext.Instance.Context)
+            {
+                SysStsgenids sysids = context.Stsgenids.OrderByDescending(q => q.Create).ToList().FirstOrDefault();
+                if(sysids != null)
+                {
+                    var oldid = int.Parse(sysids.ID.Split(separator)[2]);
+                    id = Digit2StringCreate(oldid + 1, 7);
+                }
+                else
+                {
+                    ExceptionManager.Instance.LogExceptionToFile(new SqliteExceptions.EntityNotFound("Stsgenids table is empty"));
+                }
+            }
 
-        public static string GenerateMD5Hash(string obj)
+            var newid = prefix.ToString() + separator + inerfix.ToString() + separator + id;
+            if (BasePropertiesValidator.ValidateID(newid))
+                return newid;
+            else
+                throw new BaseExceptions.IDException("Incorect id generate");
+        }
+
+        public static string Digit2StringCreate(int digit, int length)
+        {
+            var result = digit.ToString();
+            var diffLength = length - result.Length;
+            if (diffLength < 0)
+                throw new GenerateExceptions.InvalidDigitLenght("Too big number");
+
+            for(int i = 0; i < diffLength; i++)
+            {
+                result = "0" + result;
+            }
+
+            return result;
+        }
+
+        public static string GenerateSha256Hash(string obj)
         {
             var hash = string.Empty;
-            using (MD5 md5Hash = MD5.Create())
+            using (SHA256 sha256Hash = SHA256.Create())
             {
-                hash = GetMd5Hash(md5Hash, obj);
+                hash = GetSha256Hash(sha256Hash, obj);
             }
             return hash;
         }
 
-        public static bool VerifyMD5Hash(string plaintText, string hash)
+        public static bool VerifySha256Hash(string plaintText, string hash)
         {
-            using (MD5 md5Hash = MD5.Create())
+            using (SHA256 sha256Hash = SHA256.Create())
             {
-                return VerifyMd5Hash(md5Hash, plaintText, hash);
+                return VerifySha256(sha256Hash, plaintText, hash);
             }
         }
 
-        private static string GetMd5Hash(MD5 md5Hash, string input)
+        private static string GetSha256Hash(SHA256 sha256Hash, string input)
         {
-            byte[] data = md5Hash.ComputeHash(Encoding.UTF8.GetBytes(input));
+            byte[] data = sha256Hash.ComputeHash(Encoding.UTF8.GetBytes(input));
 
             StringBuilder sBuilder = new StringBuilder();
 
@@ -43,9 +91,9 @@
             return sBuilder.ToString();
         }
 
-        private static bool VerifyMd5Hash(MD5 md5Hash, string input, string hash)
+        private static bool VerifySha256(SHA256 sha256Hash, string input, string hash)
         {
-            string hashOfInput = GetMd5Hash(md5Hash, input);
+            string hashOfInput = GetSha256Hash(sha256Hash, input);
 
             StringComparer comparer = StringComparer.OrdinalIgnoreCase;
 
Index: csharp/WPF/Databases/Models/BaseEntityModel.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Databases/Models/BaseEntityModel.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Databases/Models/BaseEntityModel.cs	(date 1564038961942)
@@ -7,7 +7,7 @@
 
 namespace WPF.Databases.Models
 {
-    public class BaseEntityModel : NotifyPropertyChangedExtension, Triggerable
+    public class BaseEntityModel : NotifyPropertyChangedExtension, ITriggerable
     {
         protected string BaseID;
         protected DateTime? BaseLastUpdate;
@@ -19,7 +19,7 @@
             Context = context;
         }
 
-        public void Add()
+        public void Insert()
         {
             Context.Add(new SysStsgenids
             {
@@ -34,7 +34,7 @@
             Context.Update(this);
         }
 
-        public void Remove()
+        public void Delete()
         {
             var sts = Context.Stsgenids.Where(q => q.ID.Equals(BaseID)).FirstOrDefault();
             if (sts != null)
@@ -43,6 +43,7 @@
                 {
                     sts.Delete = DateTime.Now;
                     Context.Update(sts);
+                    Context.Remove(this);
                 }
                 else
                 {
Index: csharp/WPF/Validators/BasePropertiesValidator.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- csharp/WPF/Validators/BasePropertiesValidator.cs	(revision 36309755036bc7da98da782fa58a133ff63602c9)
+++ csharp/WPF/Validators/BasePropertiesValidator.cs	(date 1564040727878)
@@ -1,3 +1,4 @@
+using WPF.Enums;
 using WPF.Exceptions;
 
 namespace WPF.Validators
@@ -28,6 +29,16 @@
             {
                 throw new BaseExceptions.IDException("ID postfix is not number");
             }
+            return true;
+        }
+
+        public static bool ValidatePrimaryKey(string value, IDInerfixEnum tablePrefix)
+        {
+            if(ValidateID(value) && !tablePrefix.ToString().Equals(value.Split(idSeperator)[1]))
+            {
+                throw new BaseExceptions.IDException("Invalid table prefix");
+            }
+
             return true;
         }
     }
